runtime/src/internal/manifest-client.d.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  ComponentConstructor,
+  DOMComponentModule,
+  Preload
+} from './shared';
+
+export interface DOMComponentModule {
+  default: ComponentConstructor;
+  preload?: Preload;
+}
+
+export interface DOMComponent {
+	$set: (data: any) => void;
+	$destroy: () => void;
+}
+
+export interface ComponentConstructor {
```

Comment:
This should really be DOMComponentConstructor

(but I can do this in a follup up comment)

Reply:
done